sensors/mqtt_sensor.py

```diff
@@ -37,7 +37,7 @@ def setup(self):
         self._logger.debug('[MQTTSensor]: setting up sensor...')
 
         self._client = mqtt.Client(self._client_id, clean_session=True,
-                             userdata=self._userdata, protocol=self._protocol)
+                             userdata=self._userdata, protocol=getattr(mqtt, self._protocol))
```

Comment:
What should happens if the attr is missing?

Reply:
@cognifloyd Good point, I need to handle that better.  Looks like `paho.mqtt.Client()` itself defaults to `MQTTv311` but I need to handle the case where this parameter may be missing/invalid.  

Wondering if it is possible to define a list of possible values in the pack config schema?  I looked at the [pack config docs](https://docs.stackstorm.com/reference/pack_configs.html) but I don't see anything obvious in there.

Reply:
OK so I looked at the st2 code a bit and it doesn't seem like the config schema allows enforcing a specific set of strings (though correct me if I'm wrong about this) so I'll work around it in the code for now instead...

Reply:
> it doesn't seem like the config schema allows enforcing a specific set of strings (though correct me if I'm wrong about this)

You can use `enum` similar to this: https://github.com/StackStorm/st2/blob/master/contrib/examples/actions/weather.yaml#L16-L20

You'll want to modify the config schema here:

https://github.com/StackStorm-Exchange/stackstorm-mqtt/blob/b36d0718434108069aaa406a018de9b4af226e71/config.schema.yaml#L18-L22

Reply:
@cognifloyd great thanks! That was the "missing" example I was looking (in the wrong place of course) for!  I'll update the config schema to use the enum type.